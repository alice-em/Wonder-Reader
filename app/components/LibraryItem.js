import PropTypes from 'prop-types';
import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import { FaPercent } from 'react-icons/lib/fa';

import LibraryItemIcon from './LibraryItemIcon';
// import Bookmark from './Bookmark.js';

const styles = {
  bbb: {
    color: '#bbb',
  },
  percent: {
    color: '#999',
    fontSize: '14px',
  },
  TableRow: {
    cursor: 'pointer',
    fontFamily: 'Montserrat',
    fontSize: '20px',
  },
  three33: {
    color: '#333',
  },
  wide: {
    width: '10px',
  },
};

const LibraryItem = ({ basename, dirname, id, isDirectory, onRowClick }) => (
  <TableRow
    className="library-item"
    key={id}
    onClick={onRowClick}
    style={styles.TableRow}
  >
    <TableCell numeric padding="checkbox" style={styles.wide}>
      <LibraryItemIcon iconSize={30} isDirectory={isDirectory} />
    </TableCell>
    <TableCell style={styles.three33}>{basename}</TableCell>
    <TableCell numeric style={styles.bbb}>
      {dirname}
    </TableCell>
    <TableCell padding="checkbox" style={styles.wide}>
      <FaPercent style={styles.percent} />
    </TableCell>
  </TableRow>
);

LibraryItem.propTypes = {
  basename: PropTypes.string.isRequired,
  dirname: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  isDirectory: PropTypes.bool.isRequired,
  onRowClick: PropTypes.func.isRequired,
};

export default LibraryItem;
