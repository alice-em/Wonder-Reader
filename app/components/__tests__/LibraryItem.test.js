import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import LibraryItem from '../LibraryItem';

Enzyme.configure({ adapter: new Adapter() });

const props = {
  basename: 'basename',
  dirname: 'dirname',
  id: 'id',
  isDirectory: true,
  onRowClick: jest.fn(),
};

describe('LibraryItem', () => {
  it('should render', () => {
    const wrapper = shallow(<LibraryItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
