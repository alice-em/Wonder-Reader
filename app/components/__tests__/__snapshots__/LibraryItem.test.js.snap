// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryItem should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LibraryItem
    basename="basename"
    dirname="dirname"
    id="id"
    isDirectory={true}
    onRowClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "id",
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(TableCell)
          numeric={true}
          padding="checkbox"
          style={
            Object {
              "width": "10px",
            }
          }
        >
          <LibraryItemIcon
            iconSize={30}
            isDirectory={true}
          />
        </WithStyles(TableCell)>,
        <WithStyles(TableCell)
          style={
            Object {
              "color": "#333",
            }
          }
        >
          basename
        </WithStyles(TableCell)>,
        <WithStyles(TableCell)
          numeric={true}
          style={
            Object {
              "color": "#bbb",
            }
          }
        >
          dirname
        </WithStyles(TableCell)>,
        <WithStyles(TableCell)
          padding="checkbox"
          style={
            Object {
              "width": "10px",
            }
          }
        >
          <FaPercent
            style={
              Object {
                "color": "#999",
                "fontSize": "14px",
              }
            }
          />
        </WithStyles(TableCell)>,
      ],
      "className": "library-item",
      "onClick": [MockFunction],
      "style": Object {
        "cursor": "pointer",
        "fontFamily": "Montserrat",
        "fontSize": "20px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <LibraryItemIcon
            iconSize={30}
            isDirectory={true}
          />,
          "numeric": true,
          "padding": "checkbox",
          "style": Object {
            "width": "10px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "iconSize": 30,
            "isDirectory": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "basename",
          "style": Object {
            "color": "#333",
          },
        },
        "ref": null,
        "rendered": "basename",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "dirname",
          "numeric": true,
          "style": Object {
            "color": "#bbb",
          },
        },
        "ref": null,
        "rendered": "dirname",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FaPercent
            style={
              Object {
                "color": "#999",
                "fontSize": "14px",
              }
            }
          />,
          "padding": "checkbox",
          "style": Object {
            "width": "10px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "style": Object {
              "color": "#999",
              "fontSize": "14px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "id",
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(TableCell)
            numeric={true}
            padding="checkbox"
            style={
              Object {
                "width": "10px",
              }
            }
          >
            <LibraryItemIcon
              iconSize={30}
              isDirectory={true}
            />
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)
            style={
              Object {
                "color": "#333",
              }
            }
          >
            basename
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)
            numeric={true}
            style={
              Object {
                "color": "#bbb",
              }
            }
          >
            dirname
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)
            padding="checkbox"
            style={
              Object {
                "width": "10px",
              }
            }
          >
            <FaPercent
              style={
                Object {
                  "color": "#999",
                  "fontSize": "14px",
                }
              }
            />
          </WithStyles(TableCell)>,
        ],
        "className": "library-item",
        "onClick": [MockFunction],
        "style": Object {
          "cursor": "pointer",
          "fontFamily": "Montserrat",
          "fontSize": "20px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <LibraryItemIcon
              iconSize={30}
              isDirectory={true}
            />,
            "numeric": true,
            "padding": "checkbox",
            "style": Object {
              "width": "10px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "iconSize": 30,
              "isDirectory": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "basename",
            "style": Object {
              "color": "#333",
            },
          },
          "ref": null,
          "rendered": "basename",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "dirname",
            "numeric": true,
            "style": Object {
              "color": "#bbb",
            },
          },
          "ref": null,
          "rendered": "dirname",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FaPercent
              style={
                Object {
                  "color": "#999",
                  "fontSize": "14px",
                }
              }
            />,
            "padding": "checkbox",
            "style": Object {
              "width": "10px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "style": Object {
                "color": "#999",
                "fontSize": "14px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
